packages/vaadin-grid/test/array-data-provider.test.js

```diff
@@ -250,3 +251,40 @@ describe('invalid paths', () => {
     });
   });
 });
+
+describe('items with a custom data provider', () => {
+  let grid;
+  const dataProvider = (params, callback) => callback([{ value: 'foo' }], 1);
+  const items = [{ value: 'bar' }];
+
+  beforeEach(() => {
+    grid = fixtureSync(`
+      <vaadin-grid>
+        <vaadin-grid-column path="value"></vaadin-grid-column>
+      </vaadin-grid>
+    `);
+  });
+
+  it('use the items array', () => {
+    grid.dataProvider = dataProvider;
+    grid.items = items;
+    flushGrid(grid);
+    expect(getBodyCellContent(grid, 0, 0).textContent).to.equal('bar');
+    expect(grid.dataProvider).not.to.equal(dataProvider);
+  });
+
+  it('should use the data provider', () => {
+    grid.items = items;
+    grid.dataProvider = dataProvider;
+    flushGrid(grid);
+    expect(getBodyCellContent(grid, 0, 0).textContent).to.equal('foo');
+    expect(grid.items).to.be.undefined;
+  });
+
+  it('should unset the data provider', () => {
```

Comment:
I'm not sure if I totally understand what this test checks on (considering the description). 

Does it check on whether a custom provider is unset or the array data provider is unset? If the latter case, then probably it would be better to place it in [the `with items` section](https://github.com/vaadin/web-components/pull/1976/files#diff-25fcc44d6a81f4b9e6b7626f9b8f28873c65581af1b52f4ddbde64378b4adefaR21).

Reply:
True, it wasn't about a custom data provider at all. Moved to the items section.